src/xtgeo/well/_well_roxapi.py

```diff
@@ -152,84 +156,116 @@ def export_well_roxapi(
-        xtglimit = UNDEF_LIMIT
-        if self.wlogtypes[lname] == _AttrType.DISC.value:
-            isdiscrete = True
-            xtglimit = UNDEF_INT_LIMIT
-
-        if isdiscrete:
-            thelog = lrun.log_curves.create_discrete(name=lname)
+    vals = np.ma.masked_invalid(self.dataframe[logname].values)
+    vals = np.ma.masked_greater(vals, xtglimit)
+    vals = vals.astype(usedtype)
+    thelog.set_values(vals)
+
+    if isdiscrete:
+        # roxarapi requires keys to be ints, while xtgeo can accept any, e.g. strings
+        if vals.mask.all():
+            codedict = {0: "unset"}
         else:
-            thelog = lrun.log_curves.create(name=lname)
+            codedict = {
+                int(key): str(value) for key, value in self.wlogrecords[logname].items()
+            }
+        thelog.set_code_names(codedict)
 
-        values = thelog.generate_values()
 
-        if values.size != self.dataframe[lname].values.size:
-            raise ValueError("New logs have different sampling or size, not possible")
+def _roxapi_update_well(
+    self,
+    rox: Any,
+    wname: str,
+    lognames: Union[str, list[str]],
+    logrun: str,
+    trajectory: str,
+    realisation: int,
+    update_option: Optional[Literal["overwrite", "append"]] = None,
+):
+    """Assume well is to updated only with logs, new only are appended
+
+    Also, the length of arrays are not allowed not change (at least not for now).
 
-        usedtype = values.dtype
+    """
+    logger.debug("Key realisation not in use: %s", realisation)
+    if update_option not in (None, "overwrite", "append"):
+        raise ValueError(
+            f"The update_option <{update_option}> is invalid, valid "
+            "options are: None | overwrite | append"
+        )
 
-        vals = np.ma.masked_invalid(self.dataframe[lname].values)
-        vals = np.ma.masked_greater(vals, xtglimit)
-        vals = vals.astype(usedtype)
-        thelog.set_values(vals)
+    lrun = rox.project.wells[wname].wellbore.trajectories[trajectory].log_runs[logrun]
 
-        if isdiscrete:
-            # roxarapi requires keys to int, while xtgeo can accept any, e.g. strings
-            if vals.mask.all():
-                codedict = {0: "unset"}
-            else:
-                codedict = {
-                    int(key): str(value)
-                    for key, value in self._wlogrecords[lname].items()
-                }
-            thelog.set_code_names(codedict)
+    # find existing lognames in target
+    current_logs = [lname.name for lname in lrun.log_curves]
+
+    uselognames = self.lognames if lognames == "all" else lognames
+
+    if update_option is None:
+        lrun.log_curves.clear()  # clear existing logs; all will be replaced
+
+    for lname in uselognames:
+        if (
+            (update_option and update_option == "append" and lname not in current_logs)
+            or (update_option and update_option == "overwrite")
+            or update_option is None
```

Comment:
But also, these checks against `update_option` may not be needed at all since an exception will be raised at the start of the function if it is not one of the allowed ones. I think you could get away with something like

```python
for lname in uselognames:
    if update_option == "append" and lname in current_logs:
        continue
    _store_log_in_roxapi(self, lrun, lname)
```